### PR TITLE
chore: implement verification functions

### PR DESCRIPTION
**Pull request by aziyatali**

### CODE REVIEW

**Comment by aziyatali** [comment]
@Jscripter-pk  Please take a look to this. I was not able to fix this.
Basically, when I get the image and pass it as form, getting an error. 

Please let me know if you find anything useful, otherwise I will try to look for tomorrow!

Screen Shot 2023-02-09 at 10 27 51 PM

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
@aziyatali It would be better to have meeting with @MuhammadBilal789 about how he is making endpoint accept id card photo and the other thing that I would suggest here is having a seperate module for Verifications related endpoint so that we have to look specifically in one module when working with verifications

**Comment by aziyatali** [comment]
@MuhammadBilal789 Would you please review the verifications/idcare/ocr and let me know if anything is wrong?

**Comment by Jscripter-pk** [comment]
@aziyatali I added and fixed few things you just have to add you id card with idCard.jpeg name in test directory.
After Bilal provide you with further clarifications why 400/500 error is happening you can make changes accordingly then

**Comment by aziyatali** [comment]
@Jscripter-pk  Here is the working code. Also fixed the config header.
From app side, able to hit the server. Currently having issue: "please upload png, or jpg file, there are some issues with react-native-image-picker'. I think we should move to expo-image-picker since react-native-imgage-picker has MIME type issues which is buggy when we send the image over the network

**Review by Jscripter-pk — COMMENTED** [review]
Test it on app side by releasing with these changes and see if it works fine with app

**Comment by aziyatali** [comment]
> Test it on app side by releasing with these changes and see if it works fine with app

Tested and it is working fine. Please check the latest changes on app side.

**Comment by Jscripter-pk** [comment]
@aziyatali getting this error with the changes you make in app and let me tell you few things which you might be mis confusing with
```"message": "파일형식이 올바르지 않습니다. jpg, png파일을 이용해주세요.", "success": false}
```
This type error is coming because you are passing object without any image actually in formData image part 
```const formattedFile = {
      lastModified: new Date().getTime(),
      lastModifiedDate: new Date(),
      name: file?.fileName,
      size: file?.fileSize,
      type: 'image/jpg',
      webkitRelativePath: '',
    };
    ```
   The above is just a javascript object not an image/file which is supposed to be send to the endpoint that's why it give type error not because of mime type. I have told you it is not a mime type issue I tested with normal axios even with rn-image-picker library we are using. Just check the pattern thing and try to compare the data sent to the endpoint in case of bilal RN sent.

**Comment by aziyatali** [comment]
> @aziyatali getting this error with the changes you make in app and let me tell you few things which you might be mis confusing with
> 
> This type error is coming because you are passing object without any image actually in formData image part
> 
> ```
>       lastModified: new Date().getTime(),
>       lastModifiedDate: new Date(),
>       name: file?.fileName,
>       size: file?.fileSize,
>       type: 'image/jpg',
>       webkitRelativePath: '',
>     };
> ```
> 
> The above is just a javascript object not an image/file which is supposed to be send to the endpoint that's why it give type error not because of mime type. I have told you it is not a mime type issue I tested with normal axios even with rn-image-picker library we are using. Just check the pattern thing and try to compare the data sent to the endpoint in case of bilal RN sent.

[ const formattedFile = {
      lastModified: new Date().getTime(),
      lastModifiedDate: new Date(),
      uri: file?.uri,
      name: file?.fileName,
      size: file?.fileSize,
      type: file?.type,
      webkitRelativePath: ""
    }](url)
    
    Working like a magic!
    Please add url and pass actual uri and it will work!

**Comment by aziyatali** [comment]
@Jscripter-pk Please review changes that I moved from base to verifications and merge if everything is ok

File: src/types/kross-client/index.ts [hunk 1]
```diff
@@ -53,8 +53,8 @@ export type UserRegisterDto = {
   keyid: string;
   password: string;
   password2: string;
-  isBusiness: boolean;
-  isBorrower: boolean;
+  isBusiness?: boolean;
```
Comment (Jscripter-pk) [inline src/types/kross-client/index.ts:56]: Screenshot 2023-03-08 at 10 49 52 AM

**Review by Jscripter-pk — COMMENTED** [review]
Please remove the console.logs from the code and fixed the comment I leave

File: src/types/kross-client/index.ts (hunk 1 above)
Comment (aziyatali) [reply src/types/kross-client/index.ts:56]: Those are the optional values which were incorrectly put in the docs. Ijaz confirmed!

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
isBorrower and isBusiness both are required that's why I shared a screenshot with you so that you can see it clearly

**Comment by aziyatali** [comment]
> isBorrower and isBusiness both are required that's why I shared a screenshot with you so that you can see it clearly

Ijaz told those are optional params. He will update the docs.

**Pull request merged**